libslide/src/bignum/compare.rs

```diff
@@ -0,0 +1,131 @@
+use crate::bignum::Bignum;
+use std::cmp::Ordering;
+
+impl Ord for Bignum {
+    fn cmp(&self, other: &Bignum) -> Ordering {
+        let greater: Ordering;
+        let less: Ordering;
+        if self.is_neg && !other.is_neg {
+            return Ordering::Less;
+        } else if other.is_neg && !self.is_neg {
+            return Ordering::Greater;
```

Comment:
Do you want to see a way to save a branch in a somewhat convoluted manner? Great, you're in luck!
```suggestion
        if self.is_neg ^ other.is_neg {
             return other.is_neg.cmp(&self.is_neg)
        }
```